meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc

```diff
@@ -42,12 +45,12 @@ do_provide_mender_defines() {
         bbfatal "To compile U-Boot with mender you have to add 'mender-uboot' to MENDER_FEATURES_ENABLE or DISTRO_FEATURES."
     fi
 
-    if [ $(expr ${MENDER_RESERVED_SPACE_BOOTLOADER_DATA} % \( ${MENDER_PARTITION_ALIGNMENT} \* 2 \)) -ne 0 ]; then
-        bbfatal "MENDER_RESERVED_SPACE_BOOTLOADER_DATA is not an even multiple of MENDER_PARTITION_ALIGNMENT."
+    if [ $(expr ${MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK} % \( ${MENDER_PARTITION_ALIGNMENT} \* 2 \)) -ne 0 ]; then
+        bbfatal "MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK is not an even multiple of MENDER_PARTITION_ALIGNMENT."
     fi
 
-    if [ ${MENDER_BOOTENV_TOTAL_ALIGNED_SIZE} -gt ${MENDER_RESERVED_SPACE_BOOTLOADER_DATA} ]; then
-        bbfatal "BOOTENV_SIZE (${BOOTENV_SIZE}) is too big to fit two copies inside MENDER_RESERVED_SPACE_BOOTLOADER_DATA (${MENDER_RESERVED_SPACE_BOOTLOADER_DATA}) with proper alignment. Please either: 1. Increase MENDER_RESERVED_SPACE_BOOTLOADER_DATA manually and make sure it is an *even* multiple of MENDER_PARTITION_ALIGNMENT (${MENDER_PARTITION_ALIGNMENT}). -or- 2. Decrease BOOTENV_SIZE in the U-Boot recipe so that it can fit two copies inside MENDER_RESERVED_SPACE_BOOTLOADER_DATA. Please see https://docs.mender.io/troubleshooting/yocto-project-build for more information."
+    if [ ${MENDER_BOOTENV_TOTAL_ALIGNED_SIZE} -gt ${MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK} ]; then
+        bbfatal "BOOTENV_SIZE (${BOOTENV_SIZE}) is too big to fit two copies inside MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK (${MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK}) with proper alignment. Please either: 1. Increase MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK manually and make sure it is an *even* multiple of MENDER_PARTITION_ALIGNMENT (${MENDER_PARTITION_ALIGNMENT}). -or- 2. Decrease BOOTENV_SIZE in the U-Boot recipe so that it can fit two copies inside MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK. Please see https://docs.mender.io/troubleshooting/yocto-project-build for more information."
```

Comment:
I think an easier approach here, instead of introducing `MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK`, is to skip these checks if `MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART != 0`. Make sure you add a comment about why they are skipped, since this might not be obvious to the reader.

Reply:
Addressed in [force push](https://github.com/mendersoftware/meta-mender/compare/f92f8114ca70418350489e978f3ae198f578d045..f65ae24861b63bcd85e22fb9fa08236339d9a801), also updated commit comment.